pageserver/src/tenant/timeline.rs

```diff
@@ -4974,58 +4975,81 @@ impl Timeline {
 
         pausable_failpoint!("Timeline::find_gc_cutoffs-pausable");
 
-        // First, calculate pitr_cutoff_timestamp and then convert it to LSN.
-        //
-        // Some unit tests depend on garbage-collection working even when
-        // CLOG data is missing, so that find_lsn_for_timestamp() doesn't
-        // work, so avoid calling it altogether if time-based retention is not
-        // configured. It would be pointless anyway.
-        let pitr_cutoff = if pitr != Duration::ZERO {
+        if cfg!(test) {
+            // Unit tests which specify zero PITR interval expect to avoid doing any I/O for timestamp lookup
+            if pitr == Duration::ZERO {
+                return Ok(GcCutoffs {
+                    pitr: self.get_last_record_lsn(),
+                    horizon: cutoff_horizon,
+                });
+            }
+        }
+
+        // Calculate a time-based limit on how much to retain:
+        // - if PITR interval is configured, then this is our cutoff.
+        // - if PITR interval is not configured, then we do a lookup
```

Comment:
Does "not configured" mean `Duration::ZERO`? That sounds confusing. The default, if you create a new project and don't configure anything is something different, right?

Reply:
Agree that `configured` is ambiguous here.  I've changed this to "set" and cleaned up the function docstring.  f6ad06248

I think we can in future remove the default PITR in the pageserver: this policy really belongs to the control plane, and a sane default for the pageserver would be no PITR at all.